Tasks/Common/packaging-common/locationUtilities.ts

```diff
@@ -150,6 +150,26 @@ export function getWebApiWithProxy(serviceUri: string, accessToken?: string): vs
     return new vsts.WebApi(serviceUri, credentialHandler, options);
 }
 
+// This function is to apply retries generically for any unreliable network calls
+async function retryOnExceptionHelper<T>(action: () => Promise<T>, maxRetries: number, retryIntervalInMilliseconds: number): Promise<T> {
+    while (true) {
+        try {
+            return await action();
+        } catch (error) {
+            tl.debug(`Network call failed. Number of retries left: ${maxRetries}`);
```

Comment:
what would be in the "error" here. anything useful to log for the intermediate retries?

Reply:
It's mostly empty json in the cases I have seen. We are logging the last error in the caller method. Do you still want to log every error from retries?

Reply:
if it isn't structured, then i agree better not to log. it would be nice if high level result was logged. status code (or timeout), afdheader if applicable, etc.

Reply:
so long as the final result at least looks as good as it does today (it says ECONNRESET, or ETIMEDOUT, etc)

Reply:
it does, final result is same/better than what it is today.